docs/src/test.md

```diff
@@ -46,6 +46,96 @@ pytest -svv forge/test/mlir/test_ops.py::test_add
 
 > - The `-svv` flag is optional and used to display more information about the test run.
 
+## Using `@pytest.mark.parametrize`
+
+The `@pytest.mark.parametrize` decorator allows you to run a single test function with multiple sets of inputs. This is particularly useful for testing a function with various combinations of arguments and expected results.
+
+### Syntax
+
+```python
+@pytest.mark.parametrize("arg1, arg2, expected", [
+    (1, 2, 3),
+    (2, 3, 5),
+    (3, 5, 8),
+])
+def test_addition(arg1, arg2, expected):
+    assert arg1 + arg2 == expected
+```
+
+### Explanation
+- The first argument is a comma-separated string of parameter names.
+- The second argument is a list of tuples, each containing values for the parameters.
+- Pytest runs the test function once for each tuple, substituting the parameter values.
+
+## Marking Specific Parameters with `pytest.param`
```

Comment:
For example, this section can focus on explaining when, how and why we use xfail. Similar to skip